<launch>
  <!--
    Push the left frontal button labeled as 'LB' to activate cmd_vel publishing.
    Move the left stick around to control the velocity.
   -->
  <!--  smooths inputs from cmd_vel_mux/input/teleop_raw to cmd_vel_mux/input/teleop -->
  <include file="$(find turtlebot_teleop)/launch/includes/velocity_smoother.launch.xml"/>

  <param name="/joystick/dev" value="/dev/input/js1" />

  <node pkg="turtlebot_teleop" type="turtlebot_teleop_joy" name="turtlebot_teleop_joystick" output="screen">
    <param name="scale_angular" value="1.5"/>
    <param name="scale_linear" value="0.5"/>
    <param name="axis_deadman" value="4"/>
    <param name="axis_linear" value="1"/>
    <param name="axis_angular" value="0"/>
    <param name="a_button" value="5"/>
    <param name="updown_axis" value="4"/>
    <param name="rightleft_axis" value="3"/>
    <remap from="turtlebot_teleop_joystick/cmd_vel" to="teleop_velocity_smoother/raw_cmd_vel"/>
    <remap from="turtlebot_teleop_joystick/missilelaunchercmd" to="missilelaunchercmd"/>
  </node>

  <node pkg="joy" type="joy_node" name="joystick">
  </node>

</launch>
